src/tools/ScenarioMeasurement/SizeOnDisk/SizeOnDisk.cs

```diff
@@ -0,0 +1,134 @@
+        {
+            if (string.IsNullOrEmpty(scenarioName) || dirs is null || dirs.Length == 0)
+            {
+                Console.WriteLine("Usage: SizeOnDisk <dir0> <dir1> <dirN>");
+                return -1;
+            }
+
+            foreach(var dir in dirs)
+            {
+                if(!Directory.Exists(dir))
+                {
+                    Console.WriteLine($"Directory {dir} does not exist");
+                    return -1;
+                }
+            }
+            var directories = new Dictionary<string, Dictionary<string, long>>();
+            foreach(var dir in dirs)
+            {
+                FindVersionNumbers(dir);
+                directories.Add(dir, GetDirSize(dir));
+            }
+            var counters = new List<Counter>();
+            long totalSize = 0;
+            int totalCount = 0;
+            bool directoryIsTop = dirs.Length > 1; // if we were asked to log more than one directory, include the summary info as top counters.
+            foreach (var directory in directories)
+            {
+                var resultSize = directory.Value.Values.Sum();
+                var resultCount = directory.Value.Values.Count();
+                var name = RemoveVersions(directory.Key);
+                totalSize += resultSize;
+                totalCount += resultCount;
+                counters.Add(new Counter { MetricName = "bytes", TopCounter = directoryIsTop, Name = $"{RemoveVersions(directory.Key)}", Results = new[] { (double)resultSize } });
+                counters.Add(new Counter { MetricName = "count", TopCounter = directoryIsTop, Name = $"{RemoveVersions(directory.Key)}", Results = new[] { (double)resultCount } });
+                foreach (var file in directory.Value)
+                {
+                    counters.Add(new Counter { MetricName = "bytes", Name = $"{RemoveVersions(file.Key)}", Results = new[] { (double)file.Value } });
+                }
+            }
+            counters.Add(new Counter { MetricName = "bytes", Name = scenarioName, DefaultCounter = true, TopCounter = true, Results = new[] { (double)totalSize } });
+            counters.Add(new Counter { MetricName = "count", Name = scenarioName, TopCounter = true, Results = new[] { (double)totalCount } });
+            var reporter = Reporter.CreateReporter();
+            if (reporter != null)
+            {
+                var test = new Test();
+                test.Categories.Add("SizeOnDisk");
+                test.Name = scenarioName;
+                test.AddCounter(counters);
+                reporter.AddTest(test);
+                if (!String.IsNullOrEmpty(reportJsonPath))
+                {
+                    var json = reporter.GetJson();
+                    if(json != null)
+                    {
+                        File.WriteAllText(reportJsonPath, json);
+                    }
+                }
+            }
+            Console.WriteLine(reporter.WriteResultTable());
```

Comment:
Will we have a logger for this class?

Reply:
Good question. I did forget to use it here. 

I'm sort of torn about `Logger` at this point. Originally my goal was to use that abstraction to ensure that text went to console and some other persistent store, such as a log file. Given that our overall model doesn't allow for long term persistence I don't know if it is very useful to wrap up console output and do nothing else. I'm inclined to give up on it. What do you think?

Reply:
We sorted this offline. `Logger` is gonna go away eventually since it didn't end up being a thing we need.